File: sub-web/src/main/java/org/github/config/JacksonConfig.java
package org.github.config;

import java.time.LocalDateTime;

import org.springframework.boot.autoconfigure.jackson.Jackson2ObjectMapperBuilderCustomizer;
import org.springframework.boot.autoconfigure.jackson.JacksonProperties;
import org.springframework.context.annotation.Bean;
import org.springframework.context.annotation.Configuration;

import com.fasterxml.jackson.datatype.jsr310.ser.LocalDateTimeSerializer;

import static java.time.format.DateTimeFormatter.ofPattern;

@Configuration
public class JacksonConfig {
  @Bean
  public Jackson2ObjectMapperBuilderCustomizer jackson2ObjectMapperBuilderCustomizer(LocalDateTimeSerializer serializer) {
    return builder -> builder.serializerByType(LocalDateTime.class, serializer);
  }

  @Bean
  public LocalDateTimeSerializer localDateTimeSerializer(JacksonProperties prop) {
    return new LocalDateTimeSerializer(ofPattern(prop.getDateFormat()));
  }
}

File: sub-web/src/main/java/org/github/config/JacksonConfig.kt
package org.github.config

import com.fasterxml.jackson.datatype.jsr310.ser.LocalDateTimeSerializer
import org.springframework.boot.autoconfigure.jackson.Jackson2ObjectMapperBuilderCustomizer
import org.springframework.boot.autoconfigure.jackson.JacksonProperties
import org.springframework.context.annotation.Bean
import org.springframework.context.annotation.Configuration
import java.time.LocalDateTime
import java.time.format.DateTimeFormatter.ofPattern

@Configuration
class JacksonConfig {
  @Bean
  fun jackson2ObjectMapperBuilderCustomizer(localDateTimeSerializer: LocalDateTimeSerializer) = Jackson2ObjectMapperBuilderCustomizer { it.serializerByType(LocalDateTime::class.java, localDateTimeSerializer) }

  @Bean
  fun localDateTimeSerializer(prop: JacksonProperties) = LocalDateTimeSerializer(ofPattern(prop.dateFormat))
}
